### PR TITLE
Add arbiter support 

### LINKED ISSUE #46: Be able to configure replicate volume type
**Describe the feature you'd like to have.**
Today, the volume type is fixed as a replica 3 volume. It should be possible to use the parameters defined in #30 `parameters.volumeType.type` and `parameters.volumeType.replicate.*` to create the various types of replicated volumes.

**What is the value to the end user? (why is it a priority?)**
Not all users want 3-way replication. They can get significant space savings by using one of the arbiter types.

**How will we know we have a good solution? (acceptance criteria)**
- `parameters.volumeType.replicate.replicas` should determine the number of replicas for the volume, with the default remaining at 3
- `parameters.volumeType.replicate.arbiterType` should determine the type of arbiter used, with a default of `none`

### LINKED ISSUE #151: thin-arbiter volume type support
**Describe the feature you'd like to have.**
thin-arbiter support in the storage class to ensure with GCS deployment one can create a thin arbiter volume.

**What is the value to the end user? (why is it a priority?)**
https://docs.gluster.org/en/latest/Administrator%20Guide/Thin-Arbiter-Volumes/

How would the end user gain value from having this feature?
https://docs.gluster.org/en/latest/Administrator%20Guide/Thin-Arbiter-Volumes/

-----

Child of #46

### PR DESCRIPTION
**Pull request by humblec**
**Describe what this PR does**
Provide some context for the reviewer

**Is there anything that requires special attention?**
Do you have any questions? Did you do something clever?

**Related issues:**
Mention any github issues relevant to this PR

-----
Fixes #46
Fixes #151

### CODE REVIEW

**Comment by humblec** [comment]
retest this please

**Comment by amarts** [comment]
@humblec other than the rebase, is there anything more pending on this?

File: pkg/glusterfs/controllerserver.go [hunk 1]
```diff
@@ -83,17 +102,63 @@ func (cs *ControllerServer) ParseCreateVolRequest(req *csi.CreateVolumeRequest)
...
 	return &reqConf, nil
 }
 
+func checkForArbiter(vt *volumeType) error {
+	if vt.arbiterType[arbiterTypeNormal] {
+		vt.arbiterCount = 1
+		return nil
+	}
+	if vt.arbiterType[arbiterTypeThin] {
+		vt.Options["replicate.thin-arbiter"] = "127.0.0.1:/brickpath"
```
Comment (amarts) [inline pkg/glusterfs/controllerserver.go]: This has to be provided with a ENV variable or PVC option. preference is PVC.yml option, if not present ENV variable, if not present, error out.

**Comment by humblec** [comment]
@amarts this patch is more than just thin aribiter and according to the standards mentioned in SC design doc. However based on the urgency and limited requirement of thin aribiter, I am talking to @Madhu-1 and we will get thin arbiter in soon.

**Pull request closed without merging**